lib/Core/TxTree.cpp

```diff
@@ -2225,6 +2273,12 @@ TxTreeNode::TxTreeNode(
   // Inherit the abstract dependency or NULL
   dependency = new TxDependency(_parent ? _parent->dependency : 0, _targetData,
                                 _globalAddresses);
+
+  // Set speculation flag to false
```

Comment:
This lines should be guarded by an if statement which is true when in speculation mode.